### PR TITLE
Adding a PennyLane style for circuits and matplotlib plots

### PR DESCRIPTION
**Pull request by ikurecic**
Enables use of `draw_mpl` with style `'pennylane'` and `matplotlib.pyplot` with style `'pennylane.drawer.plot'` to create circuit diagrams and plots in the PennyLane style.

_Note:_ The relevant Matplotlib style sheet installs together with PennyLane and can be used independently of PennyLane, but it will not show up in the list of available styles (`matplotlib.pyplot.style.available`)
_Note2:_ This style looks best with the Quicksand Bold font (if font not seen in plots after installation, delete Matplotlib font cache file), but if the user doesn't have it installed, it will fall back on the Matplotlib default — no errors are expected to spawn
_Note3 (for testing):_ Matplotlib doesn't like `pip install -e .` and will resist you. Stick to `pip install .`

- **Circuit example:**
```
import pennylane as qml

dev = qml.device('lightning.qubit', wires=(0,1,2,3))

@qml.qnode(dev)
def circuit(x, z):
    qml.QFT(wires=(0,1,2,3))
    qml.IsingXX(1.234, wires=(0,2))
    qml.Toffoli(wires=(0,1,2))
    qml.CSWAP(wires=(0,2,3))
    qml.RX(x, wires=0)
    qml.CRZ(z, wires=(3,0))
    return qml.expval(qml.PauliZ(0))

fig, ax = qml.draw_mpl(circuit, style=’pennylane’)(1,1)
fig.show()
```

image

- **Plot example:**
```
import matplotlib.pyplot as plt
import numpy as np

plt.style.use('pennylane.drawer.plot')
for i in range(3):
    plt.plot(np.random.rand(10))
```

image

### CODE REVIEW

**Comment by josh146** [comment]
@ikurecic I think this is a really nice iterative improvement! I'll also tag @trbromley here in terms of, where this could fit into the PL Q2 roadmap

**Comment by ikurecic** [comment]
Current status:
- Plots: PennyLane palette (main three colors are the brightest blue, mid-level pink, and brightest yellow), uses Quicksand Bold font if present (if not, falls back on sans-serif and shouldn't spawn warnings).
image
- Circuits: Same color scheme as plots, but skips the yellow.
image
- Currently missing [test](https://github.com/PennyLaneAI/pennylane/pull/3950/checks?check_run_id=15711340769)

File: pennylane/plot.mplstyle
Comment (timmysilv) [inline pennylane/plot.mplstyle:1]: could we move this into `pennylane/drawer`?

File: tests/drawer/test_style.py
Comment (timmysilv) [inline tests/drawer/test_style.py:99]:
```suggestion
    """Tests the pennylane style sets ``plt.rcParams`` with correct values"""
```

**Review by timmysilv — COMMENTED** [review]
this looks great! I'd like some product people to weigh in on how we announce/inform users about the plot style in general because it's not something we've really done before, but I think it'll be awesome to have. the MPL circuits look great, I'm wondering about how to factor in a third colour as well but we don't really need it. maybe measurements? idk. I do think we shouldn't put the config at the top-level because people probably see it all the time. also do you know where that files appear when someone `pip install`s PL?

File: pennylane/plot.mplstyle
Comment (ikurecic) [reply pennylane/plot.mplstyle:1]: Good point! I've changed that now (including in the setup file and changelog), but this means having to call that style sheet in the plot function with `pennylane.drawer.plot` instead of `pennylane.plot`. I'll update the description here. (Although I'm not sure if this is more confusing?)

If you're asking where the `.mplstyle` file info lodges itself, it's just in the PennyLane install folder. If you'd like to use it without using PennyLane, you'd have to get it into your `.matplotlib/stylelib` folder.

File: doc/releases/changelog-dev.md
Comment (timmysilv) [inline doc/releases/changelog-dev.md:170]: ahh, that is a much longer thing to type only to satisfy my file organization... I might regret my decision now that I'm seeing it written out. maybe don't move it back just yet, but I'll let others weigh in on what they would prefer.

Comment (timmysilv) [reply doc/releases/changelog-dev.md:170]: oh I see your other comment, it should have been `pennylane.plot` before so it just adds one more thing. either way, one more is sometimes one too many so I'm still unsure

Comment (ikurecic) [reply doc/releases/changelog-dev.md:170]: You're totally right, I made a mistake that first time around. Yep, this is between `pennylane.plot` and `pennylane.drawer.plot`, depending on if it's nested one folder deeper or not. No way to avoid the 'plot' part (name of the style sheet) and no way to avoid the 'pennylane' part (need to refer to the package).

File: doc/code/qml_drawer.rst
Comment (josh146) [inline doc/code/qml_drawer.rst:57]:
```suggestion
.. |pls| image:: ../_static/style/pennylane_style.png
    :width: 100%
```

Comment (josh146) [reply doc/code/qml_drawer.rst:57]: @ikurecic you'll also need to add `pls` to the table below somewhere :)

File: doc/code/qml_drawer.rst
Comment (josh146) [inline doc/code/qml_drawer.rst:56]: @ikurecic has this file been added? I can't see it in the commits 🤔

**Review by josh146 — CHANGES_REQUESTED** [review]
Thanks @ikurecic! Only minor changes on my end.

> Circuits: Same color scheme as plots, but skips the yellow. (I'm not fully vibing with the color scheme, but I'm not sure we have a great option to incorporate the yellow here…)

We could have blue for single wire, yellow for multi wire, or vice versa? I also like the idea of using yellow for measurements.

Note: don't forget to also update the docstring of [`qml.draw_mpl`](https://xanaduai-pennylane--3950.com.readthedocs.build/en/3950/code/api/pennylane.drawer.draw_mpl.html) to list `pennylane` as an available style.

File: doc/code/qml_drawer.rst
Comment (ikurecic) [reply doc/code/qml_drawer.rst:56]: You're 100% right, I forgot.

**Comment by ikurecic** [comment]
Hey @timmysilv and @josh146 , if you're happy with the Matplotlib file path, I would suggest merging this as is now, because adding a PennyLane yellow for measurements in the circuit diagram would take some extra structures (not adjusting the style, but adjusting how styles work) and I don't think I have the time for that right now, unfortunately.

I'd update it the next time there's an empty block in my calendar (i.e., before the next release). :)
Sounds good?

File: pennylane/drawer/draw.py
Comment (timmysilv) [inline pennylane/drawer/draw.py:307]:
```suggestion
        style (str): visual style of plot. Valid strings are ``{'black_white', 'black_white_dark', 'sketch', 'pennylane',
```

File: pennylane/drawer/style.py
Comment (timmysilv) [inline pennylane/drawer/style.py:19]:
```suggestion
* add an entry to the private dictionary ``_style_map``.
* update the docstrings for ``use_style`` and ``draw_mpl``.
```

**Review by timmysilv — APPROVED** [review]
super excited to have this in!

File: doc/releases/changelog-dev.md
Comment (josh146) [reply doc/releases/changelog-dev.md:189]: @ikurecic up to you but you could also add the image here.

**Review by josh146 — APPROVED** [review]
Good to go from my end, once the docstring https://xanaduai-pennylane--3950.com.readthedocs.build/en/3950/code/qml_drawer.html is correctly showing the new style :)

File: doc/code/qml_drawer.rst
Comment (ikurecic) [reply doc/code/qml_drawer.rst:57]: I didn't clock that, thank you! Adding now.

File: doc/releases/changelog-dev.md
Comment (ikurecic) [reply doc/releases/changelog-dev.md:189]: I think it's fine without, but I'm adding a note about the font right here.

**Pull request merged**
